src/fsharp/FSharp.Compiler.Unittests/FscHelper.fs

```diff
@@ -0,0 +1,122 @@
+﻿namespace FSharp.Compiler.Unittests.Helpers
+
+open System
+open NUnit.Framework
+
+open Microsoft.FSharp.Compiler.Driver
+
+[<AutoOpen>]
+module NUnitHelpers =
+
+    let inline failtest message = Assert.Fail message
+    
+    let inline failtestf fmt = Printf.kprintf failtest fmt
+
+[<RequireQualifiedAccess>]
+module Check =
+    
+    let inline areEqual (expected: ^T) (actual: ^T) =
+        Assert.AreEqual(expected, actual)
+
+[<RequireQualifiedAccess>]
+module Expect =
+
+    type Recorder () =
+        let mutable xs = []
+        member recorder.Record(x) = xs <- box x :: xs
+        member recorder.Values = xs
+    
+    let inline spy f = 
+        let recorder = Recorder()
+        (recorder, fun ps -> recorder.Record(ps); f ps)
+
+    let inline once check (recorder: Recorder) =
+        match recorder.Values with
+        | [] -> failtest "expected one invocation"
+        | [x] ->
+            match x with
+            | :? _ as a -> check a
+            | _ -> invalidArg "check" (sprintf "expected function with type %s but was %s" typeof<_>.FullName (x.GetType().FullName))
+        | v -> failtestf "expected one invocation but were %d: %A" v.Length v
+
+
+    let inline never (r: Recorder) = 
+        match r.Values with
+        | [] -> ()
+        | v -> failtestf "no invocations expected but were %d: '%A'" v.Length v
+
+
+module ``fsharp compiler`` =
+    open System.IO
+    
+    open Microsoft.FSharp.Compiler.Driver
+
+    exception private FscExit of int
+
+    let private cd dir =
+        let old = System.IO.Directory.GetCurrentDirectory()
+        System.IO.Directory.SetCurrentDirectory dir
+        {new IDisposable with member x.Dispose() = System.IO.Directory.SetCurrentDirectory old }
+
+    let redirect out err =
+        Console.SetOut out
+        Console.SetError err
+        {new IDisposable with 
+            member x.Dispose() = 
+                let output = new StreamWriter(Console.OpenStandardOutput()) in output.AutoFlush <- true
+                Console.SetOut(output)
+                let error = new StreamWriter(Console.OpenStandardError()) in error.AutoFlush <- true
+                Console.SetOut(error) }
+
+    let logtomemory () = 
+        let stream = new MemoryStream()
+        let writer = new StreamWriter(stream)
+        writer.AutoFlush <- true
+        writer
+
+    let readfrom (memory: StreamWriter) = 
+        memory.BaseStream.Position <- 0L; 
+        let reader = new StreamReader(memory.BaseStream)
+        [ while reader.Peek() >= 0 do  yield reader.ReadLine() ]
+
+    type FscOutput = { 
+        warnings: string list;
+        consoleout: string list;
+        outputfile: FileInfo; }
+
+    let ``fsc`` (args: string list) (name, code) =
```

Comment:
I used backticks to make easier to understand is not the real `fsc` function. Failed. I'll use standard Module.function